backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/lsm/LiveMigrateDiskCommand.java

```diff
@@ -214,13 +217,32 @@ protected CreateSnapshotForVmParameters getCreateSnapshotParameters() {
         params.setDiskIdsToIgnoreInChecks(getImageGroupIds());
         params.setNeedsLocking(false);
         params.setEndProcedure(EndProcedure.COMMAND_MANAGED);
+        params.setDiskImagesMap(createDiskImagesMap());
+
         return params;
     }
 
     private Set<Guid> getImageGroupIds() {
         return Collections.singleton(getImageGroupId());
     }
 
+    private Map<Guid, DiskImage> createDiskImagesMap() {
+        if (FeatureSupported.isReplicateExtendSupported(getCluster().getCompatibilityVersion())) {
+            StorageDomainStatic sourceDomain = storageDomainStaticDao.get(getParameters().getSourceStorageDomainId());
+            if (sourceDomain.getStorageType().isBlockDomain()) {
+                DiskImage disk = getDiskImage();
+                long imageInitialSizeInBytes = ImagesHandler.computeDestinationImageInitialSizeInBytes(disk.getSize());
+
+                DiskImage diskParams = new DiskImage();
+                // TODO: if not empty then crashes destination (in CreateSnapshotDiskCommand::buildCreateSnapshotParameters()), since volume already exists.
+                //diskParams.setImageId(Guid.Empty);
+                diskParams.setInitialSizeInBytes(imageInitialSizeInBytes);
+
+                return Collections.singletonMap(disk.getId(), diskParams);
+            }
+        }
+        return Collections.emptyMap();
```

Comment:
A mutable map should be used instead since it's placed on the parameters, and will deserialization

Reply:
Looks like `Collections.EmptyMap`  class `implements Serializable`.
Anyway, changed to `new HashMap<>()`.